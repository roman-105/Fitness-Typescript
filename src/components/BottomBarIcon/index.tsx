import React from 'react';
import { HomeIcon, ExploreIcon, ProgressIcon, CoachIcon, MyClubIcon } from '../Icon';
import Typography from '../Typography/Typography';
import styles from '../Typography/typography-styles';
import formatMessage from 'format-message';
import { Routes } from '../../router/routes';

const lookup: { [index: string]: (color: string) => JSX.Element } = {
  [Routes.Home]: (color: string) => <HomeIcon fill={color} />,
  [Routes.Programs]: (color: string) => <ExploreIcon fill={color} />,
  [Routes.Progress]: (color: string) => <ProgressIcon fill={color} />,
  [Routes.Coach]: (color: string) => <CoachIcon fill={color} />,
  [Routes.Club]: (color: string) => <MyClubIcon fill={color} />
};

export function bottomBarIcon(name: Routes) {
  return function ({ color }: { color: string }): JSX.Element {
    return lookup[name](color);
  };
}

export function bottomBarLabel(name: Routes) {
  const tabNameLookup: { [index: string]: string } = {
    [Routes.Home]: formatMessage('Home'),
    [Routes.Programs]: formatMessage('Explore'),
    [Routes.Progress]: formatMessage('Progress'),
    [Routes.Coach]: formatMessage('Coach'),
    [Routes.Club]: formatMessage('Clubs')
  };

  return function ({ color }: { color: string }): JSX.Element {
    return (
      <Typography
        style={{ color, ...styles.label }}
        fontFamily="regular"
        fontSize={11}
        lineHeight={12}
      >
        {tabNameLookup[name]}
      </Typography>
    );
  };
}
