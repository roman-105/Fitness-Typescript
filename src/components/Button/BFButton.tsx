import React from 'react';
import { ViewStyle, TextStyle, TouchableOpacity } from 'react-native';
import Typography from '../Typography';
import { iconLookup } from '../Icon/index';
import styles from './bf-button-styles';
import theme from '../../theme';

interface BFButtonProps {
  title: string;
  onPress?: () => void;
  style?: ViewStyle;
  textStyle?: TextStyle;
  disabled?: boolean;
  wider?: boolean;
  white?: boolean;
  margin?: number;
  icon?: 'arrowRight';
}

const BFButton = ({
  title,
  onPress = () => {},
  style,
  disabled,
  wider,
  white,
  icon,
  margin = 0
}: BFButtonProps) => {
  const IconComponent = icon ? iconLookup[icon] : null;

  return (
    <TouchableOpacity
      testID={title}
      onPress={onPress}
      style={[
        styles.button,
        { margin },
        disabled && styles.disabled,
        wider && styles.wider,
        white && styles.white,
        icon && styles.withIcon,
        style
      ]}
      disabled={disabled}
    >
      <Typography
        type="h1"
        fontFamily="trebleRegular"
        style={[styles.text, white && styles.whiteText]}
      >
        {title}
      </Typography>

      {IconComponent && <IconComponent fill={theme.colors.primary.white} />}
    </TouchableOpacity>
  );
};

export default BFButton;
