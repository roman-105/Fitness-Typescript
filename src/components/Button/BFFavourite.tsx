import React, { useState } from 'react';
import { TouchableOpacity } from 'react-native';
import LinearGradient from 'react-native-linear-gradient';
import styles from './bf-favourite-styles';
import { HearthIcon } from '../Icon';
import theme from '../../theme';

const BFFavourite = (props: {
  isFavouriteItem?: boolean;
  onPress?: () => void;
  isLargeCard: boolean;
}) => {
  const [isFavourite, setIsFavourite] = useState<boolean>(props.isFavouriteItem || false);

  return (
    <TouchableOpacity
      style={[!isFavourite && styles.iconInactive, props.isLargeCard ? styles.large : styles.small]}
      onPress={() => {
        if (props.onPress) props.onPress();
        setIsFavourite(!isFavourite);
      }}
    >
      {isFavourite ? (
        <LinearGradient
          colors={theme.colors.refreshColors.brightOrangeGradient}
          style={styles.gradient}
        >
          <HearthIcon fill="white" height={10} />
        </LinearGradient>
      ) : (
        <HearthIcon fill="white" height={10} />
      )}
    </TouchableOpacity>
  );
};

export default BFFavourite;
