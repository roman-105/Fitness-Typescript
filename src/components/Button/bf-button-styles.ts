import { StyleSheet } from 'react-native';
import theme from '../../theme';

const styles: any = StyleSheet.create({
  button: {
    backgroundColor: theme.colors.secondary.powerPurple,
    padding: 8,
    justifyContent: 'center',
    alignItems: 'center'
  },
  disabled: {
    opacity: 0.5
  },
  wider: {
    padding: 16
  },
  white: {
    backgroundColor: theme.colors.primary.white
  },
  whiteText: {
    color: theme.colors.primary.asphaltGrey
  },
  text: {
    fontSize: 16,
    color: theme.colors.primary.white
  },
  withIcon: {
    flexDirection: 'row',
    justifyContent: 'space-between'
  }
});

export default styles;
