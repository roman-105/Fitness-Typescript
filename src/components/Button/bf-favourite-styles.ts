import { StyleSheet } from 'react-native';
import { convertHexToRGBA } from '../../utils';
import theme from '../../theme';

const styles: any = StyleSheet.create({
  icon: {
    backgroundColor: theme.colors.primary.orange
  },
  iconInactive: {
    backgroundColor: convertHexToRGBA(theme.colors.black, 0.6),
    paddingVertical: 10,
    paddingHorizontal: 5
  },
  large: {
    position: 'absolute',
    right: 8,
    top: 8,
    borderRadius: 20,
    zIndex: 1
  },
  small: {
    position: 'absolute',
    left: 35,
    top: 25,
    borderRadius: 100 / 2,
    zIndex: 1
  },
  gradient: {
    paddingVertical: 10,
    paddingHorizontal: 5,
    borderRadius: 100 / 2
  }
});

export default styles;
