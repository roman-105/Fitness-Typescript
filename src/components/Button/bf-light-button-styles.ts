import { StyleSheet } from 'react-native';
import theme from '../../theme';

const styles: any = StyleSheet.create({
  button: {
    backgroundColor: theme.colors.primary.white,
    justifyContent: 'center',
    alignItems: 'center',
    shadowColor: theme.colors.shadow,
    shadowOffset: { width: 2, height: 2 },
    shadowOpacity: 1,
    elevation: 2,
    paddingHorizontal: 16,
    paddingVertical: 8
  },
  disabled: {
    opacity: 0.5
  },
  textContainer: {
    flexDirection: 'row',
    alignItems: 'center',
    flex: 1,
    paddingVertical: 14
  },
  text: {
    fontSize: 16,
    color: theme.colors.primary.asphaltGrey
  },
  withIcon: {
    flexDirection: 'row',
    justifyContent: 'space-between'
  },
  aditionalInfo: {
    color: theme.colors.primary.asphaltGrey,
    opacity: 0.56,
    paddingLeft: 8
  }
});

export default styles;
