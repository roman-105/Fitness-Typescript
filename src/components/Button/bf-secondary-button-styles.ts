import { StyleSheet } from 'react-native';
import theme from '../../theme';

const styles: any = StyleSheet.create({
  button: {
    paddingTop: 24,
    paddingBottom: 24,
    backgroundColor: theme.colors.primary.white,
    flexDirection: 'row',
    alignContent: 'center',
    alignItems: 'center',
    justifyContent: 'space-between'
  },
  text: {
    marginLeft: 18,
    flex: 1
  },
  icon: {
    marginRight: 24
  },
  secondIcon: {
    marginLeft: 22
  }
});

export default styles;
