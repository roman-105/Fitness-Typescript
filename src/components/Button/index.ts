export { default as BFButton } from './BFButton';
export { default as BFSecondaryButton } from './BFSecondaryButton';
export { default as BFSelectorButton } from './selector/BFSelectorButton';
export { default as BFLightButton } from './BFLightButton';
