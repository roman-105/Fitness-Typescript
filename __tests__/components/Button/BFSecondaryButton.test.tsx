import React from 'react';
import { fireEvent, render } from '@testing-library/react-native';
import { BFSecondaryButton } from '../../../src/components/Button';

describe('BFSecondaryButton', () => {
  it('renders correctly', async () => {
    const { getByText } = render(<BFSecondaryButton title="Button" />);

    const textElement = getByText('Button');

    expect(textElement).not.toBeNull();
  });

  it('click on it triggers onPress func', async () => {
    const onPressFn = jest.fn();
    const { getByText } = render(<BFSecondaryButton title="Button" onPress={onPressFn} />);

    const textElement = getByText('Button');
    fireEvent.press(textElement);

    expect(textElement).not.toBeNull();
    expect(onPressFn).toHaveBeenCalledTimes(1);
  });

  it('snapshot', async () => {
    const { toJSON } = render(<BFSecondaryButton title="Button" />);

    expect(toJSON()).toMatchSnapshot();
  });
});
