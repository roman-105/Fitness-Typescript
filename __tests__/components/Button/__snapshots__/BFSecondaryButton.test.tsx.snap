// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BFSecondaryButton snapshot 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignContent": "center",
      "alignItems": "center",
      "backgroundColor": "#FFFFFF",
      "flexDirection": "row",
      "justifyContent": "space-between",
      "opacity": 1,
      "paddingBottom": 24,
      "paddingTop": 24,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "fontFamily": "HeadingProTreble-Heavy",
        },
        Object {
          "flex": 1,
          "marginLeft": 18,
        },
        Object {
          "fontSize": 14,
        },
        undefined,
        undefined,
        undefined,
        undefined,
      ]
    }
  >
    Button
  </Text>
</View>
`;
